src/metabase/models/database.clj

```diff
@@ -49,9 +49,12 @@
     (schedule-tasks! database)))
 
 (defn- post-select [{driver :engine, :as database}]
-  ;; TODO - this is only really needed for API responses. This should be a `hydrate` thing instead!
   (cond-> database
-    (driver/initialized? driver) (assoc :features (driver.u/features driver))))
+    (driver/initialized? driver)
+    ;; TODO - this is only really needed for API responses. This should be a `hydrate` thing instead!
+    (as-> db* ; database from outer cond->
+        (assoc db* :features (driver.u/features driver))
+        (driver/normalize-db-details driver db*))))
```

Comment:
You should probably only call `normalize-db-details` if the DB has the `:details` map... it's possible that we're fetching DB without `:details` and the various impls of the multimethod shouldn't have to deal with that situation

Reply:
Fixed.